### PR TITLE
Commands container

### PR DESCRIPTION
**Pull request by hope-yu-fivestars**
Add a status command that checks and prints the status of the aladdin-demo app endpoint and the redis connection.

### CODE REVIEW

File: helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      imagePullSecrets:
+        - name: aws
+      containers:
+      - image: {{.Values.deploy.ecr}}{{ .Chart.Name }}-commands:{{.Values.deploy.imageTag}}
+        imagePullPolicy: {{.Values.deploy.imagePullPolicy}}
+        name: {{ .Chart.Name }}-commands
+        env:
+          - name: PROJECT_NAME
+            value: {{ .Chart.Name }}
+          - name: NAMESPACE
```
Comment (agianey) [inline helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml]: i don't think you need this namespace env variable

File: helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+metadata:
+  name: {{ .Chart.Name }}-commands
+  labels:
+    project: {{ .Chart.Name }}
+    name: {{ .Chart.Name }}-commands
+    app: {{ .Chart.Name }}-commands
+    githash: {{ .Values.deploy.imageTag }}
+spec:
+  replicas: 1
+  strategy:
```
Comment (agianey) [inline helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml:12]: Do we not need the maxsurge/maxunavailable here? 
I guess maybe it doesn't matter if it's 1 replica

File: app/redis_util/redis_connection.py [hunk 3]
```diff
@@ -8,3 +8,6 @@
                 host=os.environ["REDIS_HOST"],
                 port=os.environ["REDIS_PORT"],
             )
```
Comment (agianey) [inline app/redis_util/redis_connection.py]: If you're going to use external name service for redis when we don't create, you can remove REDIS_HOST and REDIS_PORT from the configmap and get its host/port via k8s service discovery, similarly to how you are pinging aladdin demo. Doesn't need to be in this PR, but there is some minor inconsistency with how we are getting the redis endpoint vs how we are getting the demo svc endpoint.

File: app/commands_app/commands/status.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+def print_aladdin_demo_status():
+    print("pinging aladdin-demo ...")
+    host = os.environ["ALADDIN_DEMO_SERVICE_HOST"]
+    port = os.environ["ALADDIN_DEMO_SERVICE_PORT"]
+    url = "http://{}:{}/ping".format(host, port)
+    try: 
+        r = requests.get(url)
+        if r.status_code == 200:
+            print("aladdin demo endpoint ready")
```
Comment (agianey) [inline app/commands_app/commands/status.py]: maybe i'm wrong, but I don't think ready is the right word here - maybe we should print "aladdin demo ping successful" or something like that

File: app/commands_app/commands/status.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+def print_redis_status():
+    # TODO have this ping external redis when that gets added
+    print("pinging redis ...")
+    if os.environ["REDIS_CREATE"] == "false":
+        print("redis creation flag set to false, no other redis connection available at this time")
+        return
+    try: 
+        status = ping_redis()
+        print("redis connection ready {}".format(status))
```
Comment (agianey) [inline app/commands_app/commands/status.py]: similar comment as above

**Comment by agianey** [comment]
Other than some minor comments, looks good to me!

File: helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml (hunk 2 above)
Comment (hope-yu-fivestars) [reply helm/aladdin-demo/templates/aladdin-demo-commands.deploy.yaml:12]: The default is 25%, which i guess is okay for this??

**Review by agianey — APPROVED** [review]

**Pull request merged**